silk/datadog_checks/silk/check.py

```diff
@@ -0,0 +1,224 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+
+from six.moves.urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.time import get_timestamp
+
+from .events import SilkEvent
+from .metrics import BLOCKSIZE_METRICS, METRICS, READ_WRITE_METRICS
+
+EVENT_PATH = "events?timestamp__gte={}"
+
+STATE_ENDPOINT = 'system/state'
+SERVERS_ENDPOINT = 'system/servers'
+
+STATE_MAP = {'online': AgentCheck.OK, 'offline': AgentCheck.WARNING, 'degraded': AgentCheck.CRITICAL}
+
+
+class SilkCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'silk'
+
+    STATE_SERVICE_CHECK = "system.state"
+    SERVERS_SERVICE_CHECK = "server.state"
+    CONNECT_SERVICE_CHECK = "can_connect"
+    METRICS_TO_COLLECT = METRICS
```

Comment:
```suggestion
    METRICS_TO_COLLECT = dict(METRICS)
```
Can we do something like this so we aren't mutating METRICS directly? And move METRICS_TO_COLLECT inside the class?

Reply:
Yes I can change to prevent mutating `METRICS` directly. I'm not sure what you mean by move `METRICS_TO_COLLECT` inside the class? It is already in `SilkCheck`.

Reply:
Thanks! And ah sorry I meant `__init__`